pallets/phala/src/stakepool.rs

```diff
@@ -482,6 +501,10 @@ pub mod pallet {
 			// indices. (Theoretically we can enable the unbinding notification, and follow the
 			// same path as a force unbinding, but it doesn't sounds graceful.)
 			Self::remove_worker_from_pool(&worker);
+			if !pool.cd_workers.contains(&worker) {
```

Comment:
Please be aware that `remove_worker_from_pool` actually changes the PoolInfo object in the storage. At L506 you overwrite it, causing the worker not removed from the `workers` field.

How about moving the logic to manage `cd_worker` to `remove_worker_from_pool` (can change the name if the semantics doesn't match)? In this way, you can always keep the storage consistent and avoid an additional write to the storage.